add solr for AI-telemetry

RH-csaggin opened the pull request:
- adding the basic components for 3 replicas solr with 3 shard
- this has a zookeeper svc dependency to work with
- created exteralsecret

computate commented on ai-telemetry/base/solr/secret.yaml:
```diff
@@ -0,0 +1,11 @@
+    app.kubernetes.io/component: solr
+type: Opaque
+data:
+  solr-password: "ZnVjOVpWNHl3Nw=="
```
We should remove this secret, change the password, and make it an ExternalSecret I think.

RH-csaggin replied:
- https://github.com/OCP-on-NERC/nerc-ocp-config/pull/547 for Zookeeper was actually merged right? But we can not see it running on the cluster and nether on gitops.
- Yes, I did already implement the externalsecret but I had the secret leftover 😅

computate reviewed (changes requested):
We should squish the commits, maybe combine Zookeeper and Solr as one commit since the Zookeeper PR was never merged.

RH-csaggin committed:
Add Solr service for ai-telemetry on obs

- creation of resources for solr to work with zookeeper(dependency)
- creation of externalsecret for solr
- adding the service to the cluster-scope bundle

Signed-off-by: Cristiano Saggin <86964015+RH-csaggin@users.noreply.github.com>

computate commented on ai-telemetry/base/externalsecrets/solr-externalsecret.yaml:
```diff
@@ -0,0 +1,15 @@
+    name: solr-secret
+  dataFrom:
+    - extract:
+        key: $ENV/$CLUSTER/solr-password
```
Let's change this to be `key: $ENV/$CLUSTER/ai-telemetry/solr`

computate commented on ai-telemetry/overlays/externalsecrets/solr-externalsecret.yaml:
```diff
@@ -0,0 +1,9 @@
+spec:
+  dataFrom:
+    - extract:
+        key: nerc/nerc-ocp-obs/solr-secret
```
Let's change this to be `key: nerc/nerc-ocp-obs/ai-telemetry/solr`.

computate replied:
I created a vault secret that looks like this with a new password for Solr. 
![image](https://github.com/user-attachments/assets/e06f466c-6e34-4d9c-a40a-1c6aa1d2cd0b)

RH-csaggin replied:
Greate! thanks Chris!

computate commented on cluster-scope/bundles/solr/kustomization.yaml:
```diff
@@ -0,0 +1,6 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+commonLabels:
+  nerc.mghpcc.org/bundle: solr
+resources:
+- ../../base/core/namespaces/solr
```
We need to reference this bundle in `cluster-scope/overlays/nerc-ocp-obs/kustomization.yaml`
